### ISSUE TITLE
Subdomains

### ISSUE THREAD

**Issue by kelvinlockwood** [Charity-Shop-Exchange/Charity-Shop-Exchange#26]
We want to have subdomains that have a relationship with the Combo Box's "Select your region".

This means that when someone comes to the website via the subdomain the Combo Box will be pre-populated.

This might explain how we set up the subdomains > https://community.netlify.com/t/how-to-dynamically-create-a-subdomain/907/2

**Comment by globalmatt** [comment]
@kelvinlockwood So you want 1 subdomain per county, with all subdomains pointing to the same site (code)?

Might be something @Prouty89 can look at (I don't have Netlify access currently).

**Comment by kelvinlockwood** [comment]
@globalmatt yes thats right, the site would be the same, we'll just use the subdomain to do the pre-population of the Select your region box.

This issue also has a relationship with #27 and #28 - creating a _viral loop!_

I haven't added this to our Launch milestone, it can be part of the following milestone.

**Comment by globalmatt** [comment]
@kelvinlockwood As we discussed last month, adding region-specific subdomains raises a few more issues:

1. What should happen if the user picks a different region in the combo box? Presumably they should be redirected to the appropriate subdomain?

2. How to do the redirection? A simple 301 redirect or [modify window.location](https://stackoverflow.com/questions/35658252/subdomain-in-react-router)?

3. The 301 redirect will lose any data (such as genre selection) that might have been stored in localStorage since this [doesn't persist across subdomains](https://stackoverflow.com/questions/4026479/use-localstorage-across-subdomains) so this might need to be stored in domain-wide cookies instead, or possibly passed through in a query string to the new subdomain.

4. Not sure if the approach of modifying window.location would even work (worth a try though!).

**Comment by kelvinlockwood** [comment]
Good questions @globalmatt, we want to avoid unnecessarily harming the UX, what do you think @hmajid2301 , are there some workarounds?
